### PR TITLE
Refactor/data fetching

### PR DESCRIPTION
**Pull request by F-Kirchhoff**
refactor data fetching such that each page fetches the data it needs

### CODE REVIEW

File: src/app/pages/AddTopic/AddTopic.tsx [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
+      <ReturnButton to="..">
         <DoubleChevronLeft width={'24'} />
       </ReturnButton>
       <Header>Add Topic</Header>
       <TopicForm
         onSubmit={(newTopic: Topic) => {
-          onSubmit(newTopic)
-          nav('/')
+          handleTopicSubmit(newTopic)
+          nav('..')
```
Comment (PReichetanz) [inline src/app/pages/AddTopic/AddTopic.tsx:28]: Would "-1" be an idea here?

File: src/app/pages/AddTopic/AddTopic.tsx [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
         }}
-        onCancel={() => nav('/')}
+        onCancel={() => nav('..')}
```
Comment (PReichetanz) [inline src/app/pages/AddTopic/AddTopic.tsx:30]: Does this work?
```suggestion
        onCancel={() => nav(-1)}
```

**Review by PReichetanz — COMMENTED** [review]
Nice work!

**Review by phlgr — CHANGES_REQUESTED** [review]
This is quite a large PR please keep it a bit smaller next time. Nonetheless good job!

**Pull request merged**